### PR TITLE
refactor(core): add a note that hydration for i18n blocks is not yet supported

### PR DESCRIPTION
**Pull request by AndrewKushnir**
This commit updates the serialization logic to detect a situation when i18n was used for a template and throw an error to indicate that hydration for i18n blocks is not yet supported.

Hydration for i18n blocks will be added in follow up versions.

## PR Type
What kind of change does this PR introduce?

- [x] Refactoring (no functional changes, no api changes)

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/platform-server/test/hydration_spec.ts [hunk 1]
```diff
@@ -1765,5 +1767,137 @@ describe('platform-server integration', () => {
         verifyClientAndSSRContentsMatch(ssrContents, clientRootNode);
       });
     });
+
+    // Note: hydration for i18n blocks is not *yet* unsupported, so the tests
```
Comment (jessicajaniuk) [inline packages/platform-server/test/hydration_spec.ts:1771]:
```suggestion
    // Note: hydration for i18n blocks is not *yet* supported, so the tests
```

**Review by jessicajaniuk — APPROVED** [review]
LGTM other than the typo in the spec file.

File: packages/platform-server/test/hydration_spec.ts (hunk 1 above)
Comment (AndrewKushnir) [reply packages/platform-server/test/hydration_spec.ts]: Good catch! 👍

**Comment by AndrewKushnir** [comment]
[Presubmit](http://test/OCL:518074979:BASE:518075002:1679347176862:e5d31007).

**Comment by pkozlowski-opensource** [comment]
This PR was merged into the repository by commit 504bd992ce342e2557280043044da7c462c3221a.

**Pull request closed without merging**